### PR TITLE
Steehive Leader unit

### PR DESCRIPTION
**Pull request by rodolphecombe**
### Describe your pull request does
This pull request adds the Steel Dominant and its abilities. 
The Steel Dominant is a LVL4 unit whose purpose is to fill a role of Leader for the Steelhive. 
It benefits from the following abilities: 

1. Superior Life Form: this a a Leader ability that gives a 20-HP temporary boost to all allied adjacent Steelhive units at the beginning of the turn. The temporary boost is reset at the beginning of the next turn. If a boosted unit has lost more than 20 HP since it was boosted, it keeps its current HP. Otherwise, it loses only the additional HP, bringing it back to its HP from the beginning of the previous turn. 
2. Reprogram: this ability convert adjacent enemy Steelhive units to this unit's side at the end of the turn.
3. Mechanizing virus: when the unit hits with a melee attack, it has a 5%-chance to turn the non-leader enemy unit into a member of the Steelhive. The unit is transform into a LVL1 Steelhive unit. The unit type is chosen according to the usage of the unit:
    - Scout to Steel Wyrm
    - Fighter to Steel Slasher
    - Archer to Steel Impaler
    - Mixed Fighter to Steel Larva or Steel Oculus

### CODE REVIEW

File: units/steelhive/steel_dominant.cfg [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+[unit_type]
+    id=Steel Dominant
+    name= _ "Steel Dominant"
+    race=steelhive
+    gender="female"
+    image="units/steelhive/steel_dominant.png"
+    hitpoints=91
+    movement_type=steelhive_melee
+    movement=5
+    experience=240
```
Comment (knyghtmare) [inline units/steelhive/steel_dominant.cfg:11]: should be `experience=200` if `level=4`

File: units/steelhive/steel_dreadnought.cfg [hunk 2]
```diff
@@ -10,7 +10,7 @@
     experience=150
     level=3
     alignment=neutral
-    advances_to=null
+    advances_to=Steel Dominant
```
Comment (knyghtmare) [inline units/steelhive/steel_dreadnought.cfg:13]: Maybe better to not enable this unit in MP, better if it stays as a campaign thing with `[modify_unit_type]`

File: units/steelhive/steel_fideliant.cfg [hunk 3]
```diff
@@ -11,7 +11,7 @@
     experience=150
     level=3
     alignment=neutral
-    advances_to=null
+    advances_to=Steel Dominant
```
Comment (knyghtmare) [inline units/steelhive/steel_fideliant.cfg:14]: Maybe better to not enable this unit in MP, better if it stays as a campaign thing with `[modify_unit_type]`

File: units/steelhive/steel_vorpal.cfg [hunk 4]
```diff
@@ -11,7 +11,7 @@
     experience=150
     level=3
     alignment=neutral
-    advances_to=null
+    advances_to=Steel Dominant
```
Comment (knyghtmare) [inline units/steelhive/steel_vorpal.cfg:14]: Maybe better to not enable this unit in MP, better if it stays as a campaign thing with `[modify_unit_type]`

**Review by knyghtmare — CHANGES_REQUESTED** [review]
- Make it be a campaign only unit and be unlockable with `[modify_unit_type]`

File: units/steelhive/steel_fideliant.cfg [hunk 5]
```diff
@@ -15,7 +15,7 @@
     {AMLA_DEFAULT}
     cost=44
     usage=fighter
-    # wmllint: local spelling Fideliants
+	# wmllint: local spelling Fideliants
```
Comment (knyghtmare) [inline units/steelhive/steel_fideliant.cfg:18]: what happened here? this spacing showed up out of nowhere.

File: units/steelhive/steel_dominant.cfg [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+    race=steelhive
+    gender="female"
+    image="units/steelhive/steel_dominant.png"
+    hitpoints=91
+    movement_type=steelhive_melee
+    movement=5
+    experience=200
+    level=4
+    alignment=neutral
+	attacks=0
```
Comment (knyghtmare) [inline units/steelhive/steel_dominant.cfg:14]: Space/tab issues here?

**Review by knyghtmare — APPROVED** [review]

**Comment by knyghtmare** [comment]
Before I merge, some things to know:

- the max Xp of the level 3 steelhive units might be increased if the ENABLE_STEEL_DOMINANT is used
- This unit will be available for 1.18.x and 1.19.x BfW clients

File: units/steelhive/steel_fideliant.cfg (hunk 5 above)
Comment (rodolphecombe) [reply units/steelhive/steel_fideliant.cfg:18]: No idea. I tried with tab and no tab and the result is always a difference wrt the original

File: units/steelhive/steel_dominant.cfg (hunk 6 above)
Comment (rodolphecombe) [reply units/steelhive/steel_dominant.cfg:14]: Same issue as above.

**Comment by rodolphecombe** [comment]
Do you want me to add the changed XP in ENABLE_STEEL_DOMINANT to the PR?
What would be the value?

**Pull request merged**